### PR TITLE
Add mapping of production data to road network

### LINKED ISSUE #113: Match GPS coordiates from production data when inputting.
- [x] Decide max distance from production data to road segment when mapping
- [x] Find closest segment(s) from point
- [x] Write function to choose correct road segment if multiple options
- [x] Store production data with segment as foreign key

### PR DESCRIPTION
**Pull request by haavahe**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description

Simple map matching of production data to road network. Data coming in through the prod-data endpoint gets mapped to the closest road segment.

## Motivation and Context

closes #113 

## How Has This Been Tested?

Some tests for the production data endpoint are included, but is missing tests focused on the map matching methods.

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,6 +84,7 @@ Now we need to grant the database user access rights to the database we created.
 
 `GRANT ALL PRIVILEGES ON DATABASE orm TO username;`
 
+
```
Comment (Pontius1007) [inline README.md]: Feilmelding om brukeren din ikke er gjort til en superuser. Ordnet seg med denne linjen: ALTER ROLE <user_name> SUPERUSER;
Legge til?

File: backend/api/mapper/mapper.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    :return:
+    :rtype: dict
+    """
+    with connection.cursor() as cursor:
+        stmt = '''
+        WITH segment (id, distance)
+        AS
+        -- Find distance to segment and id
+        (
+          SELECT S.id AS id,
```
Comment (Pontius1007) [inline backend/api/mapper/mapper.py]: Litt forvirrende med variabelnavn. Endre til Seg eller Segment?

File: backend/api/mapper/mapper.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          ST_Distance(S.the_geom::geography,
+          ST_SetSRID(ST_MakePoint(%s, %s), 4326)::geography) AS distance
```
Comment (Pontius1007) [inline backend/api/mapper/mapper.py]: Vil gjerne ha en forklaring på hva den nye SRID-en er og hva den brukes til. Den er forskjellig fra SRID oppgitt fra NVDB. Kommentar eller wiki.

File: backend/api/mapper/mapper.py (hunk 2 above)
Comment (flamberhamber) [reply backend/api/mapper/mapper.py]: +1

File: backend/api/mapper/mapper.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    if row:
+        return {"id": row[0], "distance": row[1]}
+    else:
+        return None
+
+
+def map_to_segment(production_data):
+    """
+    Maps production data to road segments
+    :param production_data:
```
Comment (flamberhamber) [inline backend/api/mapper/mapper.py]: Er det ingen ting å si om production_data her?

File: backend/data/example_create_test_prod_data.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import backend.data.data_filter as filter
```
Comment (Pontius1007) [inline backend/data/example_create_test_prod_data.py]: Vil resultere i feilmelding. Importen er feil. data_filter as filter blir riktig.

File: backend/data/example_create_test_prod_data.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    API_password = ""
+
+
+def data_in():
+    url = 'http://localhost:8000/api/prod-data/'
+    data = filter.production_data_filter('../../../Driftsdata_SubSet_Small.geojson')
+
+    # Choose a sequence from data
+    # roads = data[121:]
+    roads = data[1:2]
```
Comment (Pontius1007) [inline backend/data/example_create_test_prod_data.py]: Send all the data please

File: backend/api/tests/test_prod_data.py [hunk 7]
```diff
@@ -147,19 +183,49 @@ def test_post_prod_data_list_authenticated_staff(self):
...
+        RoadSegment.objects.create(
+            the_geom=linestring, from_meter=1, county=1, hp=1, href=1, category=1, municipality=1, connlink=1,
+            shortform=1, medium=1, startdate=1, number=1, region=1, endnode=1, endposition=1, startnode=1,
+            startposition=1, status=1, stretchdistance=1, themecode=1, to_meter=1, typeofroad=1, roadsection=1,
+            roadsectionid=1, vrefshortform=1
+        )
+
+    def test_map_to_segment(self):
+        # Make test production data
+        production_data_out_of_range = [{"startlat": 63.387075002372903, "startlong": 10.3277250005425}]
```
Comment (flamberhamber) [inline backend/api/tests/test_prod_data.py]: Hadde det vært gunstig å teste på litt flere gps-punkter her?

File: backend/data/road_segmenting/example_roadnet_to_db.py [hunk 8]
```diff
@@ -19,16 +22,21 @@ def filter_road(road):
     :param road: A dictionary containing a road segment
     :return: A filtered dict with the road segment
     """
+    # Quick temporary fix for empty and single point coordinates
+    if len(road['geometry']['coordinates']) <= 1:
```
Comment (flamberhamber) [inline backend/data/road_segmenting/example_roadnet_to_db.py]: Denne kan kanskje fjernes nå

File: backend/api/migrations/0009_auto_20180411_1244.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        ('api', '0008_productiondata_segment'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='productiondata',
+            name='segment',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='api.DummyModel'),
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (haavahe) [reply backend/api/migrations/0009_auto_20180411_1244.py]: I did see: 0004_dummymodel_squashed_0009_auto_20180411_1244.py

File: backend/data/road_segmenting/example_roadnet_to_db.py (hunk 8 above)
Comment (haavahe) [reply backend/data/road_segmenting/example_roadnet_to_db.py]: Lar det stå foreløpig

**Comment by sardred** [comment]
The good news: I fixed Travis! It runs `python manage.py test`, it runs on PostGIS, and I turned data and road_segmenting into Python packages so the tests are run automatically by `python manage.py test`. 
The bad news: This code doesn't pass our tests. It says something like 
```
FAIL: test_split_segment_geometry_len (backend.data.road_segmenting.test_segmenting.TestSegmenting)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/IT2901-24-2018/orm/backend/data/road_segmenting/test_segmenting.py", line 52, in test_split_segment_geometry_len
    self.assertLess(count, 1)
AssertionError: 254 not less than 1
```
Look [here](https://travis-ci.org/IT2901-24-2018/orm/builds/367669055) for more info.

**Pull request merged**